tests/src/test/scala/it/unibo/scafi/test/functional/stdlib/TestBlockC.scala

```diff
@@ -0,0 +1,140 @@
+          false, false, false,
+          false, false, false
+        )).toMap)(net)
+      }
+      it("should work with nbr operations") {
+        exec(new TestProgram {
+          override def main(): Boolean = smaller(mid(), minHoodPlus(nbr(mid())))
+        }, ntimes = defaultNtimes)(net)
+
+        assertNetworkValues((0 to 8).zip(List(
+          true, false, false,
+          false, false, false,
+          false, false, true
+        )).toMap)(net)
+      }
+    }
+    describe("should support findParent") {
+      it("should work with a constant value") {
+        exec(new TestProgram {
+          override def main(): ID = findParent(1)
+        }, ntimes = defaultNtimes)(net)
+
+        assertNetworkValues((0 to 8).zip(List(
+          Int.MaxValue, Int.MaxValue, Int.MaxValue,
+          Int.MaxValue, Int.MaxValue, Int.MaxValue,
+          Int.MaxValue, Int.MaxValue, Int.MaxValue
+        )).toMap)(net)
+      }
+      it("should work with mid") {
+        exec(new TestProgram {
+          override def main(): ID = findParent(mid())
+        }, ntimes = defaultNtimes)(net)
+
+        assertNetworkValues((0 to 8).zip(List(
+          Int.MaxValue, 0, 1,
+          0, 0, 1,
+          3, 3, Int.MaxValue
+        )).toMap)(net)
+      }
+    }
+    describe("should support C") {
+      def sum: (Int, Int) => Int = _ + _
+
+      it("should produce a constant filed if every has the same potential") {
+        exec(new TestProgram {
+          override def main(): ID = C(1, sum, Int.MaxValue, 0)
+        }, ntimes = defaultNtimes)(net)
+
+        assertNetworkValues((0 to 8).zip(List(
+          Int.MaxValue, Int.MaxValue, Int.MaxValue,
+          Int.MaxValue, Int.MaxValue, Int.MaxValue,
+          Int.MaxValue, Int.MaxValue, Int.MaxValue
+        )).toMap)(net)
+      }
+      it("should accumulate on the device with the lowest potential") {
+        exec(new TestProgram {
+          override def main(): ID = C(mid(), sum, 1, 0)
+        }, ntimes = defaultNtimes)(net)
+
+        assertNetworkValues((0 to 8).zip(List(
+          8, 3, 1,
+          3, 1, 1,
+          1, 1, 1
+        )).toMap)(net)
+      }
+      it("should accumulate along a gradient - one source") {
+        net.addSensor[Boolean]("source", false)
+        net.chgSensorValue("source", Set(4), true)
+        exec(new TestProgram {
+          override def main(): ID = C(hopGradient(sense("source")), sum, 1, 0)
+        }, ntimes = defaultNtimes)(net)
+
+        assertNetworkValues((0 to 8).zip(List(
+          1, 1, 1,
+          1, 8, 1,
+          1, 1, 1
+        )).toMap)(net)
+      }
+      it("should accumulate along a gradient - two sources") {
+        net.addSensor[Boolean]("source", false)
+        net.chgSensorValue("source", Set(0,7), true)
+        exec(new TestProgram {
+          override def main(): ID = C(hopGradient(sense("source")), sum, 1, 0)
+        }, ntimes = defaultNtimes)(net)
+
+        assertNetworkValues((0 to 8).zip(List(
```

Comment:
There is an issue when expressing test cases against these kind of grid networks: sometimes, the choice of the "parent neighbour" from which collecting stuff is not clear as two neighbours happen to have the same value. Which neighbour is chosen? Do we have a criterion? It appears to depend on the order by which neighbours are folded and how ties are handled. For instance, consider a gradient with source `(0,0)`, what is the parent of device `(2,2)`? It could be equivalently `(1,2)` or `(2,1)`. A test in such a scenario tend to be fragile. So, to remove this ambiguity, we could use a network without ties. If we have ties, we must express assertions so that both possibilities are considered (i.e., in the above example, choosing `(1,2)` or `(2,1)` are both good solutions).